docs/src/snap/reference/certificates.md

```diff
@@ -0,0 +1,35 @@
+### Cluster Certificates
+
+#### 1. **CertificateAuthorities (CA)**
+
+| **Common Name**         | **Usage**                            | **Path on Disk**                     | **Used For**                              |
+|------------------------------|--------------------------------------|--------------------------------------|-------------------------------------------|
+| `kubernetes-ca`               | General Kubernetes CA               | `/etc/kubernetes/pki/ca.crt`         | Signing all Kubernetes-related certificates |
+| `kubernetes-front-proxy-ca`   | CA for front-end proxy              | `/etc/kubernetes/pki/front-proxy-ca.crt` | Signing certificates for the front-proxy |
+| `client-ca`                   | CA for client certificates          | `/etc/kubernetes/pki/client-ca.crt` | Signing certificates for the client |
+
+
+#### 2. **Certificates**
+
+| **Common Name**                       | **Usage** | **Path on Disk**                                     | **Used For**                                                     | **Signed By**               |
+|--------------------------------------------|-----------|------------------------------------------------------|------------------------------------------------------------------|-----------------------------|
+| `kube-apiserver`                           | Server    | `/etc/kubernetes/pki/apiserver.crt`                  | Securing the API server endpoint                                 | `kubernetes-ca`             |
+| `kube-apiserver-kubelet-client`            | Client    | `/etc/kubernetes/pki/apiserver-kubelet-client.crt`   | API server communication with kubelets                           | `kubernetes-ca-client`      |
```

Comment:
```suggestion
| `apiserver-kubelet-client`            | Client    | `/etc/kubernetes/pki/apiserver-kubelet-client.crt`   | API server communication with kubelets                           | `kubernetes-ca-client`      |
```

Looks like the documentation is out of date here. The docstring says kube-apiserver-kubelet-client but in reality it's set to apiserver-kubelet-client. https://github.com/canonical/k8s-snap/blob/main/src/k8s/pkg/k8sd/pki/control_plane.go#L224

Reply:
It should be `kube-apiserver-kubelet-client` per upstream docs: https://kubernetes.io/docs/setup/best-practices/certificates/#all-certificates. Perhaps we need to modify this in our codebase?

Reply:
Indeed. I would also be in favor of some kind of refactor that moves CN strings to constants, though it might get ugly with something that can change like "system:node:$hostname", so perhaps we need to think of a different solution.

Reply:
Hm, thanks both for catching that mismatch.
Note that if we change the names/path of those certificates we introduce a breaking change. Maybe we can get away with `refresh-certificates` handling both path in new versions? For an upgrade, we would then force the user to refresh the certificates, wdyt? (None of this would be part of this PR)

Reply:
or we can just live with this slight mismatch for now...

Reply:
lets use our codebase as the truth for now, thanks for the catch @eaudetcobello